app/models/vm_or_template/right_sizing.rb

```diff
@@ -31,8 +31,57 @@ module VmOrTemplate::RightSizing
 
     virtual_column :max_cpu_usage_rate_average_max_over_time_period,     :type => :float
     virtual_column :max_mem_usage_absolute_average_max_over_time_period, :type => :float
-    virtual_column :cpu_usagemhz_rate_average_max_over_time_period,      :type => :float
-    virtual_column :derived_memory_used_max_over_time_period,            :type => :float
+
+    virtual_attribute :cpu_usagemhz_rate_average_max_over_time_period,
+                      :float, :arel => (lambda do |t|
+                        col = :cpu_usagemhz_rate_average
+                        metric_rollup_table = MetricRollup.arel_table
+                        select_clause = metric_rollup_table[col].maximum
+
+                        now = Time.now.utc
+                        timestamp = (now - 30.days).utc..now
+                        tp_ids   = Thread.current[:default_time_profile_ids_cache]
+                        tp_ids ||= TimeProfile.default_time_profile(nil)
+                                              .profile_for_each_region
+                                              .pluck(:id)
+
+                        where_clause =
+                          metric_rollup_table[:time_profile_id].in(tp_ids)
+                            .and(metric_rollup_table[:capture_interval_name].eq("daily"))
+                            .and(metric_rollup_table[:timestamp].between(timestamp))
+                            .and(metric_rollup_table[:resource_type].eq("VmOrTemplate"))
+                            .and(metric_rollup_table[:resource_id].eq(t[:id]))
+
+                        t.grouping(
+                          metric_rollup_table.project(select_clause)
+                                             .where(where_clause)
+                        )
+                      end)
+    virtual_attribute :derived_memory_used_max_over_time_period,
+                      :float, :arel => (lambda do |t|
+                        col = :derived_memory_used
+                        metric_rollup_table = MetricRollup.arel_table
+                        select_clause = metric_rollup_table[col].maximum
+
+                        now = Time.now.utc
+                        timestamp = (now - 30.days).utc..now
+                        tp_ids   = Thread.current[:default_time_profile_ids_cache]
+                        tp_ids ||= TimeProfile.default_time_profile(nil)
```

Comment:
is this (`default_time_rpfile_ids_cache`?) generic enough to move into `TimeProfile`?